src/engine/GrbMatrix.h

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Johannes Herrmann (johannes.roland.herrmann@mars.uni-freiburg.de)
+
+#pragma once
+
+extern "C" {
+#include <GraphBLAS.h>
+}
+
+#include <memory>
+#include <vector>
+
+// This class wraps the functionality of the GraphBLAS object GrB_Matrix.
+// Currently only boolean matrices are supported.
+class GrbMatrix {
+ private:
+  std::unique_ptr<GrB_Matrix> matrix_;
+  static bool isInitialized_;
+
+ public:
+  GrbMatrix(size_t numRows, size_t numCols) {
+    matrix_ = std::make_unique<GrB_Matrix>();
+    auto info = GrB_Matrix_new(matrix_.get(), GrB_BOOL, numRows, numCols);
+    handleError(info);
+  }
+
+  GrbMatrix() { matrix_ = std::make_unique<GrB_Matrix>(); }
+
+  // Move constructor
+  GrbMatrix(GrbMatrix&& otherMatrix) {
+    matrix_ = std::move(otherMatrix.matrix_);
```

Comment:
Either it must be valid for a matrix to have a `nullptr` as the matrix (need to always check then), or you must reset the other pointer to again a valid empty unique_ptr.
If you use the first variant (I slightly tend towards it), this operation (along with the move assignment operator) can just be `= default`ed.

Reply:
(I would then suggest a private helper function that gets you the raw pointer but first checks that it is not null.

Reply:
I chose the first variant.